app/views/homepage/_grid_item.haml

```diff
@@ -1,11 +1,12 @@
-.home-fluid-thumbnail-grid-item
-  %div
-    -# Listing image
-    = render :partial => "layouts/grid_item_listing_image", :locals => {:listing => listing, :modifier_class => ""}
+- cache(listing) do
+  .home-fluid-thumbnail-grid-item
+    %div
+      -# Listing image
+      = render :partial => "layouts/grid_item_listing_image", :locals => {:listing => listing, :modifier_class => ""}
 
-    -# Listing author details
-    .home-fluid-thumbnail-grid-author
-      .home-fluid-thumbnail-grid-author-avatar
-        = small_avatar_thumb(listing.author, :class => "home-fluid-thumbnail-grid-author-avatar-image")
```

Comment:
I would say we cannot release with a possibility of that happening. We need to make sure we add touching from users to listings. Maybe active record model touching is enough but if not then we just need to explicitly cover all update paths to relevant users data (avatar icon, given name, family name, username, anything else?).

All in all I feel this is a place where the added complexity of handling the cache invalidation pays off because the frontpage is still quite heavy even after the recent query and db optimizations and majority of the traffic comes from frontpage. Great stuff @mporkola!